src/gwt/src/org/rstudio/studio/client/workbench/views/files/Files.java

```diff
@@ -607,6 +622,176 @@ void onCopyFilesPaneCurrentDirectory()
+   //  - the file must NOT already be open in a source editor 
+   //  - if the file is an RNotebook, then make sure that only ONE .RMarkdown file is opened for editing 
+   private void getUnopenedSelectedFiles(final CommandWithArg<List<FileSystemItem>> onCompleted)
+   {
+      final SourceColumnManager mgr = RStudioGinjector.INSTANCE.getSourceColumnManager();
+      final ArrayList<String> selectedFilePaths = new ArrayList<>(); 
+
+      // Use parallelCommandList to execute these requests as fast as possible in parallel, asynchronously
+      ParallelCommandList commandList = new ParallelCommandList(new Command()
+            {
+               @Override
+               public void execute() 
+               {
+                  final ArrayList<FileSystemItem> selectedFiles = new ArrayList<>(); 
+                  selectedFilePaths.forEach((String path) -> {
+                     selectedFiles.add(FileSystemItem.createFile(path));
+                  });
+                  onCompleted.execute(selectedFiles);
+               }
+            });
+
+
+      final List<FileSystemItem> notebooks = new ArrayList<>();
+
+      // pre-process to make sure there aren't any directories, and that already-open files do not count
+      // towards the column limit
+      for (FileSystemItem item : view_.getSelectedFiles()) 
+      {
+         if (!item.isDirectory() && !mgr.openFileAlreadyOpen(item, null))
+         {
+            TextFileType fileType = fileTypeRegistry_.getTextTypeForFile(item);
+            if (fileType.isRNotebook())
+               notebooks.add(item);
+            else
+               selectedFilePaths.add(item.getPath());
+         }
+      }
+
+      for (FileSystemItem notebook : notebooks)
+      {
+         commandList.addCommand(new SerializedCommand()
+               {
+                  @Override
+                  public void onExecute(final Command continuation)
+                  {
+                     final String rnbPath = notebook.getPath();
+                     final String rmdPath = FilePathUtils.filePathSansExtension(rnbPath) + ".Rmd";
+
+                     mgr.extractRmdFile(
+                           notebook,
+                           new ResultCallback<SourceDocumentResult, ServerError>()
+                           {
+                              @Override
+                              public void onSuccess(SourceDocumentResult doc)
+                              {
+                                 // this means the operation succeeded; add ONLY the Rmd file to the list
+                                 // if it is not already open in a source editor
+                                 final FileSystemItem rmdFile = FileSystemItem.createFile(rmdPath);
+                                 if (!selectedFilePaths.contains(rmdPath) 
+                                       && !mgr.openFileAlreadyOpen(rmdFile, null))
+                                    selectedFilePaths.add(rmdPath);
```

Comment:
```suggestion
                                 if (!selectedFilePaths.contains(rmdPath) &&
                                     !mgr.openFileAlreadyOpen(rmdFile, null))
                                    selectedFilePaths.add(rmdPath);
```
Nit: more often we put the operator at the beginning of the line rather than the end.